src/controllers/v1/drpc/DrpcController.ts

```diff
@@ -0,0 +1,106 @@
+import { Agent, utils } from '@credo-ts/core'
+import { Body, Controller, Path, Query, Post, Route, Tags, Response } from 'tsoa'
+import { injectable } from 'tsyringe'
+import type { DrpcRequestObject, DrpcResponseObject } from '@credo-ts/drpc'
+
+import { type RecordId } from '../../examples.js'
+import { NotFound, GatewayTimeout, BadGatewayError, InternalError } from '../../../error.js'
+import { RestAgent } from '../../../utils/agent.js'
+import DrpcReceiveHandler from '../../../drpc-handler/index.js'
+import { z } from 'zod'
+
+type DrpcRequestOptions = Omit<DrpcRequestObject, 'id'>
+type DrpcResponseOptions = Omit<DrpcResponseObject, 'id'>
+
+const rpcResponseParser = z.object({
+  jsonrpc: z.string(),
+  result: z.any().optional(),
+  error: z
+    .object({
+      code: z.number(),
+      message: z.string(),
+      data: z.any().optional(),
+    })
+    .optional(),
+  id: z.string(),
+})
+
+@Tags('Didcomm RPC')
+@Route('/v1/drpc')
+@injectable()
+export class DrpcController extends Controller {
+  private agent: RestAgent
+
+  public constructor(
+    agent: Agent,
+    private receiveHandler: DrpcReceiveHandler
+  ) {
+    super()
+    this.agent = agent
+  }
+
+  /**
+   * sends a request object to the connection
+   *
+   * @param connectionId the connection to send the request to
+   * @param requestOptions The request object
+   * @param timeout The timeout for receiving a response
+   */
+  @Post('/:connectionId/request')
+  @Response<NotFound>(404)
+  @Response<BadGatewayError>(502)
+  @Response<GatewayTimeout>(504)
+  public async sendRequest(
+    @Path('connectionId') connectionId: RecordId,
+    @Body() requestOptions: DrpcRequestOptions,
+    @Query('timeout') timeout = 5000
```

Comment:
Do we need a timeout/synchrony or do we stay true to more pure 'offline-first' DIDComm?

Reply:
The way the drpc API in credo works this is almost impossible. There's no way of listing drpc requests and no way of getting any information about the request except via the returned `responseListener` function. As such we must assume a short timeout between request and response. Offline first is explicitly prevented by the API design

Reply:
Hmm. I wonder if they discussed this during the development of their implementation. Will try to find a thread. Perhaps they were working within a limitation of DIDComm v1?

Reply:
According to the spec at https://github.com/hyperledger/aries-rfcs/blob/main/features/0804-didcomm-rpc/README.md, there is a required @type decorator which must equal `"@type": "https://didcomm.org/drpc/1.0/request"
`

Reply:
Yes but that's not the credo-ts api that's what's happening under the hood. Look at how `recvRequest` works https://github.com/openwallet-foundation/credo-ts/tree/main/packages/drpc.